src/components/controllers/page/index.js

```diff
@@ -42,56 +43,63 @@ export function useDescription(text) {
 	}, [text]);
 }
 
-const noop = () => {};
-
 export function usePage(route, lang) {
 	// on the server, pass data down through the tree to avoid repeated FS lookups
 	if (PRERENDER) {
-		const { content, meta } = getContentOnServer(route.path, lang);
+		const { content, html, meta } = getContentOnServer(route.path, lang);
 		return {
 			current: null,
 			content,
+			html,
 			meta,
-			loading: true, // this is important since the client will initialize in a loading state.
-			onLoad: noop
+			loading: true // this is important since the client will initialize in a loading state.
 		};
 	}
 
-	const [current, setCurrent] = useState(getContent(route));
+	const [current, setCurrent] = useState(getContentId(route));
 
 	const bootData = getPrerenderData(current);
 
 	const [hydrated, setHydrated] = useState(!!bootData);
-	const content = hydrated && bootData && bootData.content;
+	const [content, setContent] = useState(
+		hydrated && bootData && bootData.content
+	);
 
 	const [loading, setLoading] = useState(true);
 	const [isFallback, setFallback] = useState(false);
-	let [meta, setMeta] = useState(hydrated ? bootData.meta : {});
-	if (hydrated) meta = bootData.meta;
+	let [meta, setMeta] = useState(hydrated ? bootData.meta : undefined);
+	if (!meta) meta = (hydrated && bootData.meta) || {};
 
+	const lock = useRef();
 	useEffect(() => {
 		if (!didLoad) {
 			setLoading(true);
 		}
-	}, [getContent(route)]);
+		const contentId = getContentId(route);
```

Comment:
This ensures quickly switching to new routes doesn't flicker old content in. Triggering a page load invalidates any in-flight requests for content.